concent_api/core/validation.py

```diff
@@ -334,3 +336,47 @@ def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) ->
             )
 
     validate_scene_file(extra_data['scene_file'])
+
+
+def validate_that_golem_messages_are_signed_with_key(
+    public_key: bytes,
+    *golem_messages: message.base.Message,
+) -> None:
+    for golem_message in golem_messages:
+        if not is_golem_message_signed_with_key(public_key, golem_message):
+            raise Http400(
+                f'There was an exception when validating if golem_message {golem_message.__class__.__name__} is signed with '
+                f'public key {public_key}.',
+                error_code=ErrorCode.MESSAGE_SIGNATURE_WRONG,
+            )
+
+
+def validate_reject_report_computed_task(client_message: RejectReportComputedTask):
+
+    if (
+        isinstance(client_message.cannot_compute_task, message.CannotComputeTask) and
+        isinstance(client_message.task_failure, message.TaskFailure)
+    ):
+        raise GolemMessageValidationError(
+            "RejectReportComputedTask cannot contain CannotComputeTask and TaskFailure at the same time.",
+            error_code=ErrorCode.MESSAGE_INVALID,
+        )
+
+    if client_message.REASON is None:
```

Comment:
I think that `REASON` is an `Enum` **class**. It's not the content of the message. You should be checking `client_message.reason`.

Reply:
done